lib/vfs/Cargo.toml

```diff
@@ -7,23 +7,40 @@ license = "MIT"
 edition = "2018"
 
 [dependencies]
+wasmer-types = { path = "../types", version = "=3.2.0-alpha.1", default_features = false }
+wasmer-wasi-types = { path = "../wasi-types", version = "=3.2.0-alpha.1", default_features = false }
+
 libc = { version = "^0.2", default-features = false, optional = true }
 thiserror = "1"
 tracing = { version = "0.1" }
 typetag = { version = "0.1", optional = true }
 serde = { version = "1.0", default-features = false, features = ["derive"], optional = true }
-slab = { version = "0.4", optional = true }
 webc = { version = "4.0.0", optional = true }
+slab = { version = "0.4" }
+derivative = "2.2.0"
 anyhow = { version = "1.0.66", optional = true }
+async-trait = { version = "^0.1" }
+lazy_static = "1.4"
+fs_extra = { version = "1.2.0", optional = true }
+filetime = { version = "0.2.18", optional = true }
+bytes = "1"
+tokio = { version = "1", features = [ "io-util", "sync", "macros" ], default_features = false }
+pin-project-lite = "0.2.9"
+
+[dev-dependencies]
+tokio = { version = "1", features = [ "io-util", "rt" ], default_features = false }
 
 [features]
-default = ["host-fs", "mem-fs", "webc-fs", "static-fs"]
-host-fs = ["libc"]
-mem-fs = ["slab"]
-webc-fs = ["webc", "mem-fs", "anyhow"]
-static-fs = ["webc", "anyhow", "mem-fs"]
+default = ["std", "sys", "host-fs", "webc-fs", "static-fs"]
+host-fs = ["libc", "fs_extra", "filetime", "tokio/fs", "tokio/io-std"]
+webc-fs = ["webc", "anyhow"]
+static-fs = ["webc", "anyhow"]
 enable-serde = [
     "serde",
     "typetag"
 ]
 no-time = []
+std = [ "wasmer-types/std" ]
+core = [ "wasmer-types/core" ]
+sys = [ "wasmer-wasi-types/sys" ]
+js = [ "wasmer-wasi-types/js" ]
```

Comment:
We should not include nor leak wasmer types into vfs. VFS should be agnostic of wasmer logic & types

Reply:
It's agnostic but it's mostly using the various error types. 

We can have separate definitions and map between them, but that's honestly something I'd rather do post merge.

Reply:
I think this is something that can leak outside if we don't solve on this PR (specially because the features can cause things way more complex than they should)
So I'd push for solving this on this PR

Reply:
Yeah, I don't disagree... 

I'll tacke it.

Reply:
VFS was cleaned up!